src/Markdig/Helpers/ThrowHelper.cs

```diff
@@ -34,6 +35,28 @@ internal static class ThrowHelper
 
         public static void InvalidOperationException(string message) => throw new InvalidOperationException(message);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void CheckDepthLimit(int depth, bool useLargeLimit = false)
+        {
+            // Very conservative limit used to limit nesting in the final AST
+            // Used to avoid a StackOverflow in the recursive rendering process
+            const int DepthLimit = 128;
```

Comment:
Can we make this configurable? We might want to set a pretty small depth in our case